### PR TITLE
PROD-34947

### PR DESCRIPTION
**Pull request by ChristosLabrou**
PROD-34947

### CODE REVIEW

File: src/orka-builder.ts [hunk 1]
```diff
@@ -91,14 +91,14 @@ export default class OrkaBuilder {
...
-    const allowedOrigin = new RegExp('https?://(www\\.)?([^.]+\\.)?(' + allowedOrigins.join(')|(') + ')');
-
     return this.use(() =>
       cors({
         origin: (ctx: Context) => {
           const origin = ctx.request.headers.origin || ctx.request.origin;
           if (publicPrefixes.some(p => ctx.path.startsWith(p))) return '*';
-          return allowedOrigin.test(origin) ? origin : allowedOrigins[0];
+          const hostname = new URL(origin).hostname;
+          const isAllowedOrigin = allowedOrigins.some(o => hostname.endsWith(o));
```
Comment (klesgidis) [inline src/orka-builder.ts:100]: 💅 
```suggestion
          const isAllowedOrigin = allowedOrigins.some(hostname.endsWith);
```

**Review by klesgidis — APPROVED** [review]

File: test/examples/star-cors-example.test.ts [hunk 2]
```diff
@@ -29,4 +29,13 @@ describe('Star CORS examples', () => {
...
     response.headers['access-control-allow-origin'].should.eql('http://localhost:3000');
   });
+
+  it('/api triggers cors policy', async () => {
+    const response = await supertest('localhost:3000')
+      .get('/api')
+      .set('origin', 'http://localhost.foreign.com')
+      .expect(200);
+
+    response.headers['access-control-allow-origin'].should.not.eql('http://localhost.foreign.com');
```
Comment (nikostoulas) [inline test/examples/star-cors-example.test.ts:39]: Let's also assert with what this is equal

File: test/examples/star-cors-example.test.ts [hunk 3]
```diff
@@ -29,4 +29,13 @@ describe('Star CORS examples', () => {
 
     response.headers['access-control-allow-origin'].should.eql('http://localhost:3000');
   });
+
+  it('/api triggers cors policy', async () => {
```
Comment (nikostoulas) [inline test/examples/star-cors-example.test.ts:33]: Let's add one more test with a complext domain that the cors succeeds.

File: src/orka-builder.ts (hunk 1 above)
Comment (ChristosLabrou) [reply src/orka-builder.ts:100]: Doesn't work it crashes the function

File: test/examples/star-cors-example.test.ts (hunk 2 above)
Comment (ChristosLabrou) [reply test/examples/star-cors-example.test.ts:39]: Done

File: test/examples/star-cors-example.test.ts (hunk 3 above)
Comment (ChristosLabrou) [reply test/examples/star-cors-example.test.ts:33]: Done

File: test/examples/star-cors-example.test.ts [hunk 4]
```diff
@@ -29,4 +29,23 @@ describe('Star CORS examples', () => {
...
+    response.headers['access-control-allow-origin'].should.eql('localhost');
+  });
+
+  it('/api/example returns access-control-allow-origin that contains the subdomain', async () => {
+    const response = await supertest('localhost:3000')
+      .get('/api/example')
+      .set('origin', 'http://some.deep.subdomain.localhost')
+      .expect(200);
+
+    response.headers['access-control-allow-origin'].should.eql('http://some.deep.subdomain.localhost');
```
Comment (nikostoulas) [inline test/examples/star-cors-example.test.ts:49]: ❓ Why does this work? It shouldn't work right? Would it work with the previous implementation?

File: test/examples/star-cors-example.test.ts [hunk 5]
```diff
@@ -29,4 +29,32 @@ describe('Star CORS examples', () => {
...
+  });
+
+  it('/api/example blocks deep subdomains', async () => {
+    const response = await supertest('localhost:3000')
+      .get('/api/example')
+      .set('origin', 'http://some.very.deep.subdomain.localhost:3000')
+      .expect(200);
+
+    response.headers['access-control-allow-origin'].should.eql('localhost:3000');
+  });
```
Comment (nikostoulas) [inline test/examples/star-cors-example.test.ts:59]: Can we add another one when this is supported if in allowedOrigins the configuration is *.localhost:3000?

Comment (ChristosLabrou) [reply test/examples/star-cors-example.test.ts:59]: Done. Added '*.lvh.me' in config to not break other tests that rely on localhost

**Review by nikostoulas — APPROVED** [review]

**Comment by nikostoulas** [comment]
@tomtoump changing that wouldn't be backwards compatible. It would also cause bugs in various areas.

**Pull request merged**